### PR TITLE
fix subscription name not take effect

### PR DESCRIPTION
**Pull request by KevinyhZou**
#457 bug fix the subscription name not task effect in flink sql job

### CODE REVIEW

**Comment by syhily** [comment]
Tks for your contribution. It seems like a mistake by our maintainer.
LGTM for me.

@jianyun8023 Can you confirm why you comment on [this line](https://github.com/streamnative/pulsar-flink/blob/1366373082b04ddcd66b11801133414f84d7e558/pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/FlinkPulsarSource.java#L599)?

**Review by syhily — APPROVED** [review]

**Comment by syhily** [comment]
I'll merge this PR after @jianyun8023 confirming my question above.

File: pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/PulsarOptions.java
Comment (nlu90) [inline pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/PulsarOptions.java:58]: could you keep it consistent with other options here?

**Review by nlu90 — CHANGES_REQUESTED** [review]

**Comment by KevinyhZou** [comment]
> Tks for your contribution. It seems like a mistake by our maintainer. LGTM for me.
> 
> @jianyun8023 Can you confirm why you comment on [this line](https://github.com/streamnative/pulsar-flink/blob/1366373082b04ddcd66b11801133414f84d7e558/pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/FlinkPulsarSource.java#L599)?

OK

File: pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/PulsarOptions.java
Comment (KevinyhZou) [reply pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/PulsarOptions.java:58]: the pulsar reader  config now seems changes to subscriptionRolePrefix, if keep it 'subscription-role-prefix', exception throws like 
image
from the log I see this config show be 'subscriptionRolePrefix'

Comment (nlu90) [reply pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/PulsarOptions.java:58]: Thanks for the clarification. These config inconsistencies are causing a lot of confusions.

We will see how to work on cleaning them later

**Review by nlu90 — APPROVED** [review]

File: pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/PulsarOptions.java
Comment (KevinyhZou) [reply pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/PulsarOptions.java:58]: OK

**Review by jianyun8023 — CHANGES_REQUESTED** [review]
@KevinyhZou The subscription name is not here to set the Pulsar Reader prefix, it is only used to access the current consumption location. If you need to set the Reader's prefix, you can set `'properties.pulsar.reader.subscription-role-prefix' = 'bcd123-'` in SQL

@syhily @nlu90 I do not recommend merging this modification. If you have to use the subscription name automatically set to the Reader's subscription prefix, the method to get the subscription name `getSubscriptionName()` does not satisfy this need, it gets the subscriber or ``pulsar-flink-$uuid``.

```
protected String getSubscriptionName() {
        if (startupMode == StartupMode.EXTERNAL_SUBSCRIPTION) {
            checkNotNull(externalSubscriptionName);
            return externalSubscriptionName;
        } else {
            return "flink-pulsar-" + uuid.toString();
        }
}
```

**Comment by KevinyhZou** [comment]
> 'properties.pulsar.reader.subscription-role-prefix

@jianyun8023 then the scan.startup.mode = 'external-subscription' will be unuseful , and use 'properties.pulsar.reader.subscription-role-prefix' instead? but this is not describe in the document

**Comment by syhily** [comment]
OK, I'll close this PR. But I don't think this design is acceptable.

**Pull request closed without merging**
